framework/test/unit/modules/interoperability/mainchain/commands/submit_mainchain_cross_chain_update.spec.ts

```diff
@@ -724,171 +725,209 @@ describe('SubmitMainchainCrossChainUpdateCommand', () => {
-				context.ccm.sendingChainID,
-				BigInt(0),
-				CCMStatusCode.OK,
-				codec.encode(sidechainTerminatedCCMParamsSchema, {
-					chainID: context.ccm.receivingChainID,
-					stateRoot: chainAccount.lastCertificate.stateRoot,
-				}),
-			);
-		});
+				interopMod['internalMethod'].sendInternal = sendInternalMock;
 
-		it('should revert the state and terminate the sending chain if beforeCrossChainMessageForwarding fails', async () => {
-			(
-				(ccMethods.get('token') as BaseCCMethod).beforeCrossChainMessageForwarding as jest.Mock
-			).mockRejectedValue('error');
-			jest.spyOn(context.eventQueue, 'createSnapshot').mockReturnValue(99);
-			jest.spyOn(context.stateStore, 'createSnapshot').mockReturnValue(10);
-			jest.spyOn(context.eventQueue, 'restoreSnapshot');
-			jest.spyOn(context.stateStore, 'restoreSnapshot');
+				chainAccount = await command['stores']
+					.get(ChainAccountStore)
+					.get(context, context.ccm.receivingChainID);
+			});
 
-			const terminateChainInternalMock = jest.fn();
-			interopMod['internalMethod'].terminateChainInternal = terminateChainInternalMock;
+			it('should terminate the chain when receiving chain is not live', async () => {
+				jest.spyOn(interopMod['internalMethod'], 'terminateChainInternal');
 
-			await expect(command['_forward'](context)).resolves.toBeUndefined();
+				await expect(command['_forward'](context)).resolves.toBeUndefined();
 
-			expect(terminateChainInternalMock).toHaveBeenCalledWith(
-				expect.anything(),
-				context.ccm.sendingChainID,
-			);
-			expect(command['events'].get(CcmProcessedEvent).log).toHaveBeenCalledWith(
-				expect.anything(),
-				context.ccm.sendingChainID,
-				context.ccm.receivingChainID,
-				{
-					ccm: context.ccm,
-					code: CCMProcessedCode.INVALID_CCM_BEFORE_CCC_FORWARDING_EXCEPTION,
-					result: CCMProcessedResult.DISCARDED,
-				},
-			);
-			expect(context.eventQueue.restoreSnapshot).toHaveBeenCalledWith(99);
-			expect(context.stateStore.restoreSnapshot).toHaveBeenCalledWith(10);
+				expect(interopMod['internalMethod'].terminateChainInternal).toHaveBeenCalledWith(
+					expect.anything(),
+					context.ccm.receivingChainID,
+				);
+			});
+
+			it('should call sendInternal', async () => {
+				await expect(command['_forward'](context)).resolves.toBeUndefined();
+
+				expect(sendInternalMock).toHaveBeenCalledWith(
+					expect.anything(),
+					EMPTY_FEE_ADDRESS,
+					MODULE_NAME_INTEROPERABILITY,
+					CROSS_CHAIN_COMMAND_SIDECHAIN_TERMINATED,
+					context.ccm.sendingChainID,
+					BigInt(0),
+					CCMStatusCode.OK,
+					codec.encode(sidechainTerminatedCCMParamsSchema, {
+						chainID: context.ccm.receivingChainID,
+						stateRoot: chainAccount.lastCertificate.stateRoot,
+					}),
+				);
+			});
+
+			runBounceRelevantCommonTests();
 		});
 
-		it('should add ccm to receiving chain outbox and log event when valid', async () => {
+		it('should add ccm to receiving chain outbox and log event when live', async () => {
```

Comment:
Here, it should also check that it updated escrows correctly, i.e., it called `beforeCrossChainMessageForwarding` (which is not the case with current specs, if I understand well).